pkg/mount/mount.go

```diff
@@ -702,6 +784,16 @@ func New(
 	return nil, ErrUnsupported
 }
 
+// bindMountPath return the path used to bind mount to a user given mount path.
+// This allows us to chattr the path in case we cannot chattr the original mount path.
+func bindMountPath(path string) (string, error) {
+	pathSlice := strings.Split(path, "/")
+	if len(pathSlice) == 0 {
+		return "", fmt.Errorf("mount path cannot be empty")
+	}
+	return filepath.Join(volume.MountBase, pathSlice[len(pathSlice)-1]), nil
```

Comment:
The source for bind mount should have a prefix like `/var/lib/osd/mounts/readonly/<id>`. Docker volume driver mounts the volumes on volume.MountBase/<id>